firebase_admin/dynamic_links.py

```diff
@@ -169,47 +147,96 @@ def from_json(cls, platform, event, count):
-            raise ValueError('Count: {} must be a non negative int'.format(count))
-        self._count = count
 
+class StatOptions(object):
+    def __init__(self, duration_days):
+        self.duration_days = duration_days
+
+    @property
+    def duration_days(self):
+        return self._duration_days
+
+    @duration_days.setter
+    def duration_days(self, duration_days):
+        if (isinstance(duration_days, bool)
+                or not isinstance(duration_days, int)
+                or duration_days < 1):
+            raise ValueError('duration_days must be positive integer (got {})'
+                             .format(duration_days))
+        self._duration_days = duration_days
+
+class _LinksService(object):
+    """Provides methods for the Firebase dynamic links interaction"""
 
-def _validate_url(url):
-    if not isinstance(url, six.string_types) or not url.startswith('https://'):
-        raise ValueError('Url must be a string and begin with "https://".')
+    INTERNAL_ERROR = 'internal-error'
 
-def _validate_stat_options(options):
-    if not isinstance(options, StatOptions):
-        raise ValueError('Options must be of type StatOptions.')
-    if (isinstance(options.duration_days, bool)
-            or not isinstance(options.duration_days, int)
-            or options.duration_days < 1):
-        raise ValueError('duration_days: {} must be positive int'.format(options.duration_days))
+    def __init__(self, app):
+        self._client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_LINKS_BASE_URL)
+        self._timeout = app.options.get('httpTimeout')
+        self._request_string = '{0}/linkStats?durationDays={1}'
+
+    def _format_request_string(self, short_link, options):
+        days = options.duration_days
+        # Complaints about the named second argument needed to replace "/"
+        #pylint: disable=redundant-keyword-arg
+        url_quoted = urllib.parse.quote(short_link, safe='')
+        #pylint: enable=redundant-keyword-arg
+        return self._request_string.format(url_quoted, days)
+
+    def get_stats(self, short_link, stat_options):
+        """Returns the LinkStats of the requested short_link for the duration set in options"""
+        if(not isinstance(short_link, six.string_types)
+           or not short_link.startswith('https://')):
+            raise ValueError('short_link must be a string and begin with "https://".')
+        if not isinstance(stat_options, StatOptions):
+            raise ValueError('stat_options must be of type StatOptions.')
+
+        request_string = self._format_request_string(short_link, stat_options)
+        try:
+            resp = self._client.body('get', request_string)
+        except requests.exceptions.RequestException as error:
+
+            self._handle_error(error)
+        else:
+            link_event_stats_dict = resp.get('linkEventStats', [])
```

Comment:
It looks like this is actually a list. So may be just `link_event_stats` (without _dict)?

Reply:
yeah, I guess i meant "...dicts" but it is cleaner without